fix(PeriphDrivers): Resolve incorrect DMA request in DMA-based SPI transactions for all parts

JeonChangmin opened the pull request:
### Description
This pull request addresses an issue in the DMA-based SPI transaction implementation where the DMA request would not be executed under specific conditions.

### Problem Details
- **Issue**: When `MXC_SPI_MasterTransactionDMA` is called with `txData == NULL` and `rxData != NULL` in `SPI_WIDTH_STANDARD` (full duplex) mode, the function sets `reqselTx` to `-1`. However, in `MXC_SPI_RevA1_MasterTransactionDMA`, `MXC_SPI_RevA1_TransSetup` may additionally set `txData` based on the request. This leads to the [use of an invalid request selector](https://github.com/analogdevicesinc/msdk/blob/e1cae5bbb0bed43a754e487a99995dd85ca37b6a/Libraries/PeriphDrivers/Source/SPI/spi_reva1.c#L1019), causing the DMA to not be executed and resulting in an infinite wait state.

- **Call Stack**:
  - [MXC_SPI_MasterTransactionDMA](https://github.com/analogdevicesinc/msdk/blob/e1cae5bbb0bed43a754e487a99995dd85ca37b6a/Libraries/PeriphDrivers/Source/SPI/spi_me17.c#L420-L448)
    - Determines `reqselTx` and `reqselRx` based on the device and data presence.
  - [MXC_SPI_RevA1_MasterTransactionDMA](https://github.com/analogdevicesinc/msdk/blob/e1cae5bbb0bed43a754e487a99995dd85ca37b6a/Libraries/PeriphDrivers/Source/SPI/spi_reva1.c#L931)
    - Uses `MXC_SPI_RevA1_TransSetup` to configure the request, potentially modifying `req` to include both Tx and Rx data.
  - [MXC_SPI_RevA1_TransSetup](https://github.com/analogdevicesinc/msdk/blob/e1cae5bbb0bed43a754e487a99995dd85ca37b6a/Libraries/PeriphDrivers/Source/SPI/spi_reva1.c#L732-L745)
    - Adjusts the request to comply with SPI protocol, potentially setting both Tx and Rx data.

### Solution
- Always pass valid `reqselRx` and `reqselTx` values from `MXC_SPI_MasterTransactionDMA`.
- Configure DMA in `MXC_SPI_RevA1_MasterTransactionDMA` after executing `MXC_SPI_RevA1_TransSetup`.
> Solution Rationale: By ensuring valid `reqselRx` and `reqselTx` values are always passed from `MXC_SPI_MasterTransactionDMA`, the `MXC_SPI_RevA1_MasterTransactionDMA` function can correctly configure the DMA settings based on the request. This prevents the DMA from not being executed due to invalid request selectors. Additionally, the `MXC_SPI_RevA1_MasterTransactionDMA` function always checks the null status of `rxData` and `txData` before using the request selectors, ensuring that no other issues arise from this change.

### Additional Fixes
- Applied the same fix to `MXC_SPI_SlaveTransactionDMA`.
- Corrected potentially swapped `reqselRx` and `reqselTx` values in [spi_me11.c](https://github.com/analogdevicesinc/msdk/blob/e1cae5bbb0bed43a754e487a99995dd85ca37b6a/Libraries/PeriphDrivers/Source/SPI/spi_me11.c#L275) and [spi_me18.c](https://github.com/analogdevicesinc/msdk/blob/e1cae5bbb0bed43a754e487a99995dd85ca37b6a/Libraries/PeriphDrivers/Source/SPI/spi_me18.c#L464-L499) files. Please confirm if this change was intended. ([fix in this PR](https://github.com/analogdevicesinc/msdk/pull/1059/commits/049b11eeef3864ecb332ad281f60fe180c3c142c))

### Tests
- Tested the changes on a MAX78000 device using `spi_me17.c` as a basis.

### Checklist Before Requesting Review
- [x] PR Title follows correct guidelines.
- [x] Description of changes and all other relevant information.
- [x] Provide info on any relevant functional testing/validation.

Jake-Carter reviewed (changes requested):
Thank you for your very nice PR.

Can confirm you've caught some copy/paste typos on the mismatched reqsel values.  

The AI87 values are also swapped if you wouldn't mind fixing that as well.

JeonChangmin commented:
Thanks for the review!
I fixed the cases where `spi_num` and `reqselTx/Rx` didn't match as you suggested.
There are more than AI87, so I fixed all of them, but please let me know if there is something else you intended.

sihyung-maxim commented on Libraries/PeriphDrivers/Source/SPI/spi_ai87.c:
```diff
@@ -403,34 +403,19 @@ int MXC_SPI_MasterTransactionDMA(mxc_spi_req_t *req)
-        }
+    switch (spi_num) {
+    case 0:
+        reqselTx = MXC_DMA_REQUEST_SPI1TX;
```
I believe the original implementation was correct. SPI0 and SPI1 are swapped in several places as an unfortunate result of incorrect documentation, and it's been stuck like that ever since to prevent breaking projects... It's why we have messes like this for the ME17, AI85, and AI87. 
![image](https://github.com/analogdevicesinc/msdk/assets/75494566/3c080c73-420e-443d-b8d5-4b675004f450)

I say we fix this even if it might break existing projects. Usually, instance 0 is mapped to the lowest peripheral address, and then everything goes in ascending order:
1. SPI0 (APB) - 0x40046000
2. SPI1 (APB) - 0x40047000
3. SPI[n] (APB) - 0x4004m000 
4. SPI[n+1] (AHB) - 0x400BE000
5. SPI[n+2] (AHB) - 0x400BE400

However, the ME17, AI85, and AI87 doesn't follow this:
1. SPI0 (AHB) - 0x400BE000
2. SPI1 (APB) - 0x40046000

This has caused confusion, but it'll be a major breaking change to swap the SPI instance names. Instead, we can fix the `MXC_SPI_GET_*` definitions in the {device}.h file where SPI0 would be associated with index 0, and SPI1 with index 1. This would still be a breaking change, but it would at least help keep all the indexing aligned with the correct SPI names.

What do you think? @Jake-Carter & @lorne-maxim

Jake-Carter reviewed (approved):
Thanks @JeonChangmin, no this is good.  Looks like the same mistake got copy/pasted across the drivers since the ME12.
